ilxutils/ilxutils/cli.py

```diff
@@ -0,0 +1,395 @@
+                    super_data = super_data['data']
+                    if success:
+                        data[pred].append({'id': super_data['id'], 'ilx': _obj})
+                    else:
+                        return self.test_check('Your superclass ILX ID '
+                                                + _obj + ' does not exist.')
+            elif pred == 'existing_ids':  # FIXME need to autogenerate curies from a map
+                iris = [d['iri'] for d in data[pred]]
+                if obj not in iris:
+                    if 'http' not in obj:
+                        return self.test_check('exisiting id value must \
+                                               be a uri containing "http"')
+                    data[pred].append({
+                        'curie': self.qname(obj),
+                        'iri': obj,
+                        'preferred': '0'
+                    })
+                data = self.preferred_change(data)
+            else:
+                return self.test_check(pred + ' Has slipped through the cracks')
+        return data
+
+    # FIXME: need to sql all curie to iri mappings and put them here
+    def qname(self, iri):
+        return 'dummy:' + iri.rsplit('/', 1)[-1]
+
+    def preferred_change(self, data):
+        ranking = [
+            'CHEBI',
+            'NCBITaxon',
+            'COGPO',
+            'CAO',
+            'DICOM',
+            'UBERON',
+            'NLX',
+            'NLXANAT',
+            'NLXCELL',
+            'NLXFUNC',
+            'NLXINV',
+            'NLXORG',
+            'NLXRES',
+            'NLXSUB'
+            'BIRNLEX',
+            'SAO',
+            'NDA.CDE',
+            'PRO',
+            'NIFEXT',
+            'ILX',
+        ]
+        mock_rank = ranking[::-1]
+        score = []
+        old_pref_index = None
+        for i, d in enumerate(data['existing_ids']):
+            if int(d['preferred']) == 1:
+                old_pref_index = i
+            if d.get('curie'):
+                pref = d['curie'].split(':')[0]
+                if pref in mock_rank:
+                    score.append(mock_rank.index(pref))
+                else:
+                    score.append(-1)
+            else:
+                score.append(-1)
+
+        new_pref_index = score.index(max(score))
+        new_pref_iri = data['existing_ids'][new_pref_index]['iri']
+        if new_pref_iri.rsplit('/', 1)[0] == 'http://uri.interlex.org/base':
+            if old_pref_index:
+                if old_pref_index != new_pref_index:
+                    return data
+        for e in data['existing_ids']:
+            e['preferred'] = 0
+        data['existing_ids'][new_pref_index]['preferred'] = 1
+        return data
+
+    def add_entity(self, rdf_type, superclass, label, definition=None):
+
+        # Checks if you inputed the right type
+        accepted_types = ['term', 'cde', 'annotation', 'relationship', 'fde']
```

Comment:
These are the interlex types. We should probably add a translation layer between, for example, `term` and `owl:Class`.